src/middlewares/validateUserAddress.js

```diff
@@ -23,34 +17,42 @@ const validateUserAddress = [
   check('street_address')
     .isLength({ min: 1 })
     .withMessage('street address cannot be empty')
-    .isLength({ max: 50 })
+    .isLength({ max: 100 })
+    .withMessage('Street address should be at most 100 characters')
     .trim(),
 
   check('city')
     .isLength({ min: 1 })
-    .withMessage('please add your city')
-    .isAlpha()
-    .withMessage('please add a valid city')
+    .withMessage('City cannot be empty')
     .isLength({ max: 50 })
     .withMessage('maximum length of characters reached'),
 
   check('state')
     .isLength({ min: 1 })
-    .withMessage('please add your state')
-    .isAlpha()
-    .withMessage('please add a valid state')
+    .withMessage('State cannot be empty')
     .isLength({ max: 50 })
+    .withMessage('State cannot exceed 50 characters')
     .trim(),
 
   check('country')
     .isLength({ min: 1 })
-    .withMessage('please add your country')
-    .isAlpha()
-    .withMessage('please add your country')
+    .withMessage('Country cannot be empty')
     .isLength({ max: 50 })
+    .withMessage('Country cannot exceed 50 characters')
     .trim(),
 
   (req, res, next) => handleErrors(req, res, next),
 ];
 
-export default validateUserAddress;
+export const checkIfUserAddressExists = async (req, res, next) => {
+  const { id } = req.user;
+  try {
+    const dbOrder = await getItemByAttribute('users_address', 'user_id', id);
+    if (dbOrder.rows.length) {
+      return formatResponse(res, { error: 'You already have an address' }, 400);
+    }
+    return next();
+  } catch (error) {
+    return formatResponse(res, { error: 'An error occurred' }, 500);
+  }
+};
```

Comment:
User should be able to add any amount of address they want or at least a high max say 20